src/pipeline.cpp

```diff
@@ -137,7 +155,11 @@ void openxc::pipeline::publish(openxc_VehicleMessage* message,
 void openxc::pipeline::sendMessage(Pipeline* pipeline, uint8_t* message,
         int messageSize, MessageClass messageClass) {
     sendToUsb(pipeline, message, messageSize, messageClass);
-    sendToUart(pipeline, message, messageSize, messageClass);
+	#ifdef TELIT_HE910_SUPPORT
```

Comment:
Yeah you've got it. The reason this is necessary is because the existing devices (RN42) attached to the UART, once "initialized", are in "live" mode with the remote (phone) and don't execute any more device commands. Once initialized they give up the physical interface to the application code via a "connected" or "ready" flag so the application can interact with the remote.

The Telit interacts with the remote (server) via socket commands (all socket send and receive data is wrapped up in socket send and receive commands) so the device layer is needed at all times. Although a live mode is also possible with the Telit, socket commands are nice because they relax the timing requirements for the MCU. If we stop reading from a socket mid-transfer we aren't going to lose data (unless the TCP connection times out). Admittedly this comes with a potential penalty on throughput due to additional overhead.

More generally the separation of the physical interface from the application let's us give the device controller autonomy to manage the device and execute device commands as needed, independent of the application logic (e.g. find out why a connection dropped and try to restart it, while ignoring or buffering data the application is trying to send to the remote). Actually in this case the application calls the server APIs, which then call into the device controller, although I haven't separated these two properly.